### PR TITLE
remove TODO in test/e2e/framework/providers/gce/ingress.go

### PR DESCRIPTION
**Pull request by yuxiaobo96**
Signed-off-by: yuxiaobo <yuxiaobogo@163.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:#86763

**Special notes for your reviewer**:

The TODO in the function GetFirewallRuleOrError is related to the function GetFirewallRule. But the function GetFirewallRule is not used, so remove this function and modify the function GetFirewallRuleOrError to GetFirewallRule.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/framework/providers/gce/ingress.go [hunk 1]
```diff
@@ -570,20 +570,9 @@ func (cont *IngressController) GetFirewallRuleName() string {
...
-// GetFirewallRule returns the firewall used by the IngressController.
-// Causes a fatal error incase of an error.
-// TODO: Rename this to GetFirewallRuleOrDie and similarly rename all other
-// methods here to be consistent with rest of the code in this repo.
-func (cont *IngressController) GetFirewallRule() *compute.Firewall {
-	fw, err := cont.GetFirewallRuleOrError()
-	framework.ExpectNoError(err)
-	return fw
-}
-
```
Comment (neolit123) [inline test/e2e/framework/providers/gce/ingress.go]: according to github, this function is unused, so the change seems fine.

File: test/e2e/framework/providers/gce/ingress.go [hunk 2]
```diff
@@ -570,20 +570,9 @@ func (cont *IngressController) GetFirewallRuleName() string {
...
 // GetFirewallRuleOrError returns the firewall used by the IngressController.
 // Returns an error if that fails.
-// TODO: Rename this to GetFirewallRule when the above method with that name is renamed.
-func (cont *IngressController) GetFirewallRuleOrError() (*compute.Firewall, error) {
+func (cont *IngressController) GetFirewallRule() (*compute.Firewall, error) {
```
Comment (neolit123) [inline test/e2e/framework/providers/gce/ingress.go]: this is a good change as there is no need to have `OrError` for a function that returns an `error`.

/lgtm
/approve

**Comment by neolit123** [comment]
please fix this error:
> Errors from golint:
test/e2e/framework/providers/gce/ingress.go:573:1: comment on exported method IngressController.GetFirewallRule should be of the form "GetFirewallRule ..."

/priority backlog

**Comment by yuxiaobo96** [comment]
@neolit123 
Thanks for your review.
done

File: test/e2e/framework/providers/gce/ingress.go (hunk 2 above)
Comment (neolit123) [reply test/e2e/framework/providers/gce/ingress.go]: the function still needs an export comment on top of it?

```
// GetFirewallRule returns the firewall used by the IngressController.
// Returns an error if that fails.
func (cont *IngressController) GetFirewallRule() (*compute.Firewall, error) {
...
```

Comment (yuxiaobo96) [reply test/e2e/framework/providers/gce/ingress.go]: This comment is on line 573 and is correct.

Comment (yuxiaobo96) [reply test/e2e/framework/providers/gce/ingress.go]: I don't understand why the display here is wrong.

**Comment by oomichi** [comment]
unrelated test is failed.
The change itself is good for me.

/retest
/lgtm

**Comment by tanjunchen** [comment]
/lgtm
i think it is good

**Pull request merged**